src/Microsoft.AspNetCore.OData/Extensions/ODataServiceProviderConfigExtenions.cs

```diff
@@ -34,7 +34,7 @@ public static void SetDefaultQuerySettings(this IServiceProvider serviceProvider
                 throw Error.InvalidOperation(SRResources.MissingODataServices, nameof(DefaultQuerySettings));
             }
 
-            querySettings = defaultQuerySettings;
+            querySettings.CopySettings(defaultQuerySettings);
```

Comment:
So we are overwriting whatever settings were obtained from Dependency Injection with the settings passed in to SetDefaultQuerySettings. Is this the expected behavior?

Note that we are copying MaxTop from the default settings returned from DI *after* we have already applied semantics based on the value passed in (see line 26).  Should we move the logic on line 26-29 to *after* we do copysettings?

Reply:
Yes. Looking at the name of the extension method it should reset the default query settings with the current object. 

Moved the three lines after copying.